### ISSUE TITLE
Query url resolving by filters (+replace optional http with https for url resolving)

### ISSUE THREAD

**Issue by FlorianSteenbuck** [uBlockOrigin/uMatrix-issues#178]
### URL(s) where the issue occurs
everywhere

### Describe the issue
Their is currently no way to trigger the adurl page which leads to a resolving of the possible redirect urls. And let somehow filters trigger this page. This would be nice feature for mostly newsletter or search order email urls which often leads you into a tracking or non http link. Also some sites do include links to non http links. I know the non http thing seems to only be my flagship but for at least users that really using newsletters this is for anti tracking a good feature.

### Notes
Can you just add for every link a with https option (just nice to have) ?

**Comment by gwarser** [comment]
This helps?

https://github.com/uBlockOrigin/uMatrix-issues/wiki/How-to-get-past-%22uMatrix-has-prevented-the-following-page-from-loading%22#bypass-redirects

**Comment by llacb47** [comment]
I think they want a filter to like `||http:$document` to show the https version url on the strict blocking page...?

**Comment by uBlock-user** [comment]
It's already a feature albeit "+replace optional http with https for url resolving"

**Comment by gorhill** [comment]
I don't understand the request here -- I can't accept any feature requests at this point (see how little time I have to work on this project), and feature request unaccompanied by steps to reproduce with real world cases are even less desireable.

**Comment by gwarser** [comment]
Maybe this will solve your problem https://www.eff.org/https-everywhere ?

Two people does not understand what you mean - invalid.

**Comment by FlorianSteenbuck** [comment]
Hope this solves the misunderstanding :pray: 
TLDR @llacb47 is correct

Anyways here is the way how to reproduce something that you should understand just by normal browser knowledge:

- auto complete http:// issue
    - open a browser that got a visible autocomplete for protcol of a url (Firefox)
    - type in any domain that do not got a record in you browser history (as their is no top level support check you can type in everything that can be split at least once with the delimiter ".") 
    - you can see now that the autocomplete of firefox add the unencrypted http protocol as prefix this a basic problem in every environment where the connection can be intercepted, which is in case of the internet everytime the case
- the newsletter problem
    - search for a crappy newsletter that contains http links or redirect links with ids
    - click a link of the newsletter 
        - you end in a tracking link that got id to identify your email (Most users just want the content)
        - or in a unsafe http link
- the crappy http link problem
    - open https://tabula.technology/
    - click on the link to the Knight Foundation
    - and you land a http page
    - maybe with redirects maybe without

immowelt.de got problems with the newletter
tabula got the crappy http link problem on their side
every user got the problem while type in a new domain name

This shouldn't be a problem of uBlock, uMatrix and so on. This should be a problem of the browser, currently their is no default protocol prefix setting.

HTTPS Everywhere is a nice problem solver, but do not solve it in a general way. Your adurl page seems to got a solution even for web tracking. It currently can only be triggered for one domain. For backfall reasons I need this adurl page for selective http -> https, I do not just redirect this directly because their are still http site that I want access to.

**Comment by gwarser** [comment]
https://addons.mozilla.org/en-US/firefox/search/?q=https&type=extension

https://addons.mozilla.org/en-US/firefox/search/?q=clear%20url%20redirect&type=extension
